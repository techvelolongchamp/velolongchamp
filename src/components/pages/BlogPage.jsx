import React from 'react'
import { Link, graphql, useStaticQuery } from 'gatsby'
import styled from 'styled-components'

import Layout from '../Layout'
import Header from '../Header'
import Head from '../Head'
import ThirdarySection from '../sections/ThirdarySection'

const SubTitle = styled.p`
  font-size: ${({ theme }) => theme.fonts.large};
  margin: ${({ theme }) => theme.spacing.l} 0;
  color: ${({ theme }) => theme.colors.brandTertiary};
  font-weight: 600;
`

const Posts = styled.ol`
  list-style-type: none;
  margin: 0;
`

const Post = styled.li`
  margin: 20px 0;

  a {
    background: ${({ theme }) => theme.colors.offWhite};
    color: ${({ theme }) => theme.colors.defaultColor};
    display: block;
    padding: 20px;
    text-decoration: none;
    border-radius: 5px;
  }

  a:hover {
    background: ${({ theme }) => theme.colors.darkerOffWhite};
  }

  h2 {
    font-size: ${({ theme }) => theme.fonts.large};
  }
`

const PublishDate = styled.p`
  margin: ${({ theme }) => theme.spacing.xs} 0;
`

const Extract = styled.p`
  color: ${({ theme }) => theme.colors.grey};
  font-size: ${({ theme }) => theme.fonts.medium};
  font-style: italic;
`

const BlogPage = ({ lng }) => {
  const data = useStaticQuery(graphql`
    query {
      allMarkdownRemark(
        sort: { fields: frontmatter___date, order: DESC }
        limit: 10
      ) {
        edges {
          node {
            frontmatter {
              title
              slug
              date(locale: "fr", formatString: "Do MMMM YYYY")
            }
            excerpt
          }
        }
      }
    }
  `)

  return (
    <Layout lng={lng}>
      <Header noScroll lng={lng} />
      <Head title="Blog" url="/blog" lng={lng} />
      <ThirdarySection title="Blog" useH1>
        <SubTitle>Liste des articles publi√©s :</SubTitle>
        <Posts>
          {data.allMarkdownRemark.edges.map(post => (
            <Post key={post.node.frontmatter.slug}>
              <Link to={`/blog/${post.node.frontmatter.slug}`}>
                <h2>{post.node.frontmatter.title}</h2>
                <PublishDate>{post.node.frontmatter.date}</PublishDate>
                <Extract>{post.node.excerpt}</Extract>
              </Link>
            </Post>
          ))}
        </Posts>
      </ThirdarySection>
    </Layout>
  )
}

export default BlogPage
