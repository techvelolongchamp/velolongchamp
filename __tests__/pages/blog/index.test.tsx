import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import BlogPage from '../../../pages/blog/index'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const posts = [
  {
    id: '1',
    title: 'title1',
    date: '2021-03-24T15:48:12.516Z',
    excerpt: 'excerpt1',
    _sys: { filename: 'slug1' },
  },
  {
    id: '2',
    title: 'title2',
    date: '2021-03-10T15:48:12.516Z',
    excerpt: 'excerpt2',
    _sys: { filename: 'slug2' },
  },
]

describe('BlogPage unit tests', () => {
  it('should render correctly', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container } = render(<BlogPage posts={posts} />)

    expect(container).toMatchSnapshot()
  })
})
