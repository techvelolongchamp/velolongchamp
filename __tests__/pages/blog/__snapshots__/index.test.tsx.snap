// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage unit tests should render correctly 1`] = `
.c24 {
  background: #f7f7F7;
  border-top: 1px solid #f7f7F7;
  color: #999999;
  padding: 10px 0 5px;
  text-align: center;
}

.c24 a {
  color: #999999;
}

.c24 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c25 {
  background: #0d171b;
  bottom: 0;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  color: white;
  font-size: 14px;
  padding: 10px;
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;
}

.c25 div {
  margin-top: 10px;
}

.c25 a {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  padding: 5px;
}

.c25 button {
  background: #419a1c;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  outline: done;
  padding: 5px;
  width: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-top: 80px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  border-bottom-left-radius: 30px;
  border-top-left-radius: 30px;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 40px;
  font-weight: 100;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: 0;
  top: 120px;
  z-index: 10;
}

.c1 svg {
  margin: 10px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: -200px;
  top: 120px;
  z-index: 10;
  width: 200px;
  font-size: 16px;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  border-bottom-left-radius: 10px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 10px 20px 0;
  padding: 5px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #419a1c;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 120px;
  border-radius: 5px;
}

.c3:hover {
  color: #f7f7F7;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin: 0 5px;
}

.c15 {
  background: none;
  border: none;
  outline: none;
  color: black;
  padding: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #999999;
  cursor: pointer;
  padding: 0 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:hover {
  color: #666666;
}

.c5 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 15;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 15px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c13 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c9 {
  height: 70px;
  padding: 5px 0;
}

.c17 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c16 {
  min-height: 600px;
}

.c18 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

.c19 {
  font-size: 16px;
  margin: 20px 0;
  color: #9e9577;
  font-weight: 600;
}

.c20 {
  list-style-type: none;
  margin: 0;
}

.c21 {
  background: #f7f7F7;
  border-radius: 5px;
  color: #0d171b;
  cursor: pointer;
  display: block;
  margin: 20px 0;
  padding: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c21:hover {
  background: #ededed;
}

.c21 h2 {
  font-size: 16px;
}

.c22 {
  margin: 5px 0;
}

.c23 {
  color: #666666;
  font-size: 14px;
  font-style: italic;
}

@media (max-width:640px) {
  .c4 {
    margin-top: 110px;
  }
}

@media screen and (max-width:640px) {
  .c10 {
    font-size: 14px;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c11 {
    padding: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c13 {
    font-size: 14px;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
        data-testid="show-alert"
        id="show-alert"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
          />
          <line
            x1="12"
            x2="12"
            y1="9"
            y2="13"
          />
          <line
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </div>
      <div
        class="c2"
      >
        Signaler un accident ou problème sur l'anneau
        <a
          class="c3"
          data-testid="notice-alert"
          href="https://docs.google.com/forms/d/e/1FAIpQLScSoqJxlSnDBXA1Qk_d81RNj8V5sRJ3CI0OOhO84R2b1L3sQQ/viewform"
          id="notice-alert"
          rel="noopener noreferrer"
          target="_blank"
          title="Formulaire"
        >
          Formulaire
        </a>
      </div>
    </div>
    <main
      class="c4"
    >
      <header
        class="c5"
        id="header"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <a
              class="c8"
              data-testid="title-link"
              href="/"
              id="logo-link"
            >
              <img
                alt="logo"
                class="3 c9"
                src="/logo.png"
              />
            </a>
            <nav>
              <ul
                class="c10"
                data-testid="navbar"
              >
                <li>
                  <a
                    class="c11"
                    href="/"
                    id="nav-link-home"
                  >
                    A propos
                  </a>
                </li>
                <li>
                  <a
                    class="c11"
                    href="/blog"
                    id="nav-link-blog"
                  >
                    Actualités
                  </a>
                </li>
                <li>
                  <a
                    class="c11"
                    href="/calendar"
                    id="nav-link-calendar"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="c12"
          >
            <a
              class="2 c13"
              href="/join"
              id="join-link"
            >
              Adhérer gratuitement
            </a>
            <div
              class="c14"
            >
              <a
                class="c15"
                data-testid="select-en"
                href="/"
                id="select-en"
              >
                EN
              </a>
            </div>
          </div>
        </div>
      </header>
      <section
        class="c16"
      >
        <div
          class="c17"
        >
          <h1
            class="c18"
            data-testid="h1-title"
          >
            Actualités
          </h1>
          <p
            class="c19"
          >
            Liste des articles publiés :
          </p>
          <ol
            class="c20"
          >
            <a
              href="/blog/slug1"
            >
              <li
                class="c21"
              >
                <h2>
                  title1
                </h2>
                <p
                  class="c22"
                >
                  24 mars 2021
                </p>
                <p
                  class="c23"
                >
                  excerpt1
                </p>
              </li>
            </a>
            <a
              href="/blog/slug2"
            >
              <li
                class="c21"
              >
                <h2>
                  title2
                </h2>
                <p
                  class="c22"
                >
                  10 mars 2021
                </p>
                <p
                  class="c23"
                >
                  excerpt2
                </p>
              </li>
            </a>
          </ol>
        </div>
      </section>
    </main>
    <footer
      class="c24"
    >
      <p>
        ©Copyright 2019
         - 2023
         - Vélo Longchamp -
         
        Tous droits réservés
      </p>
      <p>
        -
      </p>
      <p
        id="legal-link"
      >
        <a
          href="/legal"
        >
          Mentions légales
        </a>
      </p>
    </footer>
    <div
      class="c25"
    >
      <p>
        En poursuivant votre navigation sur notre site, vous acceptez l'utilisation de cookies afin de nous permettre d'améliorer votre expérience utilisateur.
      </p>
      <div>
        <button
          data-testid="accept-cookie"
          id="accept-cookie"
        >
          OK
        </button>
        <a
          href="/legal"
        >
          En savoir plus
        </a>
      </div>
    </div>
  </div>
</div>
`;
